File: csharp/lesson 2/ValidParentheses.cs
// https://leetcode.com/problems/valid-parentheses/

class Solution 
{
    public bool IsValid(String s) 
    {
        Stack<char> stack = new Stack<char>();        
        foreach (char c in s)
        {
            if (c == '(' || c == '{' || c == '[')
            {
                stack.Push(c);
            } 
            else 
            {
                if (stack.Count == 0) return false;
                char o = stack.Pop();
                if (o == '{' && c != '}' || o == '(' && c != ')' || o == '[' && c != ']' ) return false;
            }
        }
        return stack.Count == 0;
    }
}

File: python/lesson 2/valid_parentheses.py
# https://leetcode.com/problems/valid-parentheses/

class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if  c == '(' or c == '{' or c == '[':
                stack.append(c)
            else:
                if len(stack) == 0: return False
                o = stack.pop()
                if o == '{' and c != '}' or o == '(' and c != ')' or o == '[' and c != ']': return False;        
        return len(stack) == 0